engine/src/dskmac.cpp

```diff
@@ -4711,18 +4712,20 @@ struct MCMacDesktop: public MCSystemInterface, public MCMacSystemService
         }
         return "unknown";
 #endif /* MCS_getmachine_dsk_mac */
-        static Str255 machineName;
-        long response;
-        if ((errno = Gestalt(gestaltMachineType, &response)) == noErr)
-        {
-            GetIndString(machineName, kMachineNameStrID, response);
-            if (machineName != nil)
-            {
-                p2cstr(machineName);
-                return MCStringCreateWithNativeChars((const char_t *)machineName, MCCStringLength((const char *)machineName), r_string);
-            }
-        }
-        return MCStringCopy(MCNameGetString(MCN_unknown), r_string);
+
+		// PM-2015-07-21: [[ Bug 15623 ]] machine() returns "unknown" in OSX because of Gestalt being deprecated
+		size_t t_len = 0;
+		sysctlbyname("hw.model", NULL, &t_len, NULL, 0);
+
+		if (t_len)
+		{
+			char *t_model = (char *)malloc(t_len * sizeof(char));
+			sysctlbyname("hw.model", t_model, &t_len, NULL, 0);
+
+			return MCStringCreateWithCStringAndRelease(t_model, r_string);
```

Comment:
@runrevpanos: Careful here - AndRelease semantics do not free the buffer if an error is thrown. Therefore you need to do:
```
if (!MCStringCreateWithCStringAndRelease(t_model, r_string))
{
  free(t_model);
  return false;
}
return true;
```